api.c

```diff
@@ -0,0 +1,1216 @@
+					percentage = 1e2 * thisforward / forwardedsum * (overTime[i].total - (overTime[i].cached + overTime[i].blocked)) / overTime[i].total;
+				}
+				else
+					percentage = 0.0;
+
+				if(istelnet[*sock])
+					ssend(*sock, " %.2f", percentage);
+				else
+					pack_float(*sock, (float) percentage);
+			}
+
+			// Avoid floating point exceptions
+			if(overTime[i].total > 0)
+				// Forward count for destination "local" is cached + blocked normalized by total:
+				percentage = 1e2 * (overTime[i].cached + overTime[i].blocked) / overTime[i].total;
+			else
+				percentage = 0.0;
+
+			if(istelnet[*sock])
+				ssend(*sock, " %.2f\n", percentage);
+			else
+				pack_float(*sock, (float) percentage);
+		}
+	}
+
+	if(debugclients)
+		logg("Sent overTime forwarded data to client, ID: %i", *sock);
+}
+
+void getClientID(int *sock)
+{
+	if(istelnet[*sock])
+		ssend(*sock,"%i\n", *sock);
+	else
+		pack_int32(*sock, *sock);
+
+	if(debugclients)
+		logg("Sent client ID to client, ID: %i", *sock);
+}
+
+void getQueryTypesOverTime(int *sock)
+{
+	int i, sendit = -1;
+	for(i = 0; i < counters.overTime; i++)
+	{
+		validate_access("overTime", i, true, __LINE__, __FUNCTION__, __FILE__);
+		if((overTime[i].total > 0 || overTime[i].blocked > 0))
+		{
+			sendit = i;
+			break;
+		}
+	}
+
+	if(sendit > -1)
+	{
+		for(i = sendit; i < counters.overTime; i++)
+		{
+			validate_access("overTime", i, true, __LINE__, __FUNCTION__, __FILE__);
+
+			double percentageIPv4 = 0.0, percentageIPv6 = 0.0;
+			int sum = overTime[i].querytypedata[0] + overTime[i].querytypedata[1];
+
+			if(sum > 0) {
+				percentageIPv4 = 1e2*overTime[i].querytypedata[0] / sum;
+				percentageIPv6 = 1e2*overTime[i].querytypedata[1] / sum;
+			}
+
+			if(istelnet[*sock])
+				ssend(*sock, "%i %.2f %.2f\n", overTime[i].timestamp, percentageIPv4, percentageIPv6);
+			else {
+				pack_int32(*sock, overTime[i].timestamp);
+				pack_float(*sock, (float) percentageIPv4);
+				pack_float(*sock, (float) percentageIPv6);
+			}
+		}
+	}
+
+	if(debugclients)
+		logg("Sent overTime query types data to client, ID: %i", *sock);
+}
+
+void getVersion(int *sock)
+{
+	const char * commit = GIT_HASH;
+	const char * tag = GIT_TAG;
+
+	if(strlen(tag) > 1) {
+		if(istelnet[*sock])
+			ssend(*sock, "version %s\ntag %s\nbranch %s\ndate %s\n", GIT_VERSION, tag, GIT_BRANCH, GIT_DATE);
+		else {
+			pack_str32(*sock, GIT_VERSION);
+			pack_str32(*sock, (char *) tag);
+			pack_str32(*sock, GIT_BRANCH);
+			pack_str32(*sock, GIT_DATE);
+		}
+	}
+	else {
+		char hash[8];
+		// Extract first 7 characters of the hash
+		strncpy(hash, commit, 7); hash[7] = 0;
+
+		if(istelnet[*sock])
+			ssend(*sock, "version vDev-%s\ntag %s\nbranch %s\ndate %s\n", hash, tag, GIT_BRANCH, GIT_DATE);
+		else {
+			char *hashVersion = calloc(5 + strlen(hash), sizeof(char));
+			sprintf(hashVersion, "vDev-%s", hash);
+
+			pack_str32(*sock, hashVersion);
+			pack_str32(*sock, (char *) tag);
```

Comment:
There is a compiler warning otherwise:
```
api.c: In function 'getVersion':
api.c:1007:22: warning: passing argument 2 of 'pack_str32' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
    pack_str32(*sock, tag);
                      ^~~
In file included from api.c:12:0:
api.h:41:6: note: expected 'char *' but argument is of type 'const char *'
 void pack_str32(int sock, char *string);
      ^~~~~~~~~~
```